### PR TITLE
PIM-3900: Add PriceCollectionAttributeAdder

### PR DESCRIPTION
**Pull request by willy-ahva**
| Q                    | A
| -------------------- | ---
| Bug fix?             | n
| New feature?         | n
| BC breaks?           | n
| CI currently passes? | y
| Tests pass?          | y
| Scenarios pass?      | y
| Checkstyle issues?*  | n
| PMD issues?**        | n
| Changelog updated?   | n
| Fixed tickets        | PIM-3900
| DB schema updated?   | n
| Migration script?    | n
| Doc PR               |

### CODE REVIEW

File: features/update/add_price.feature [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      | BOX1    | [{"type": "add_data", "field": "price", "data": [{"data": "12.4", "currency": "EUR"}], "locale": "fr_FR", "scope": null}] | {"values": {"price": [{"locale": "fr_FR", "scope": null, "value": [{"data": 12.4, "currency": "EUR"}]}]}}                                 |
+      | BOX1    | [{"type": "add_data", "field": "price", "data": [{"data": 5, "currency": "EUR"}], "locale": "fr_FR", "scope": null}]      | {"values": {"price": [{"locale": "fr_FR", "scope": null, "value": [{"data": 5, "currency": "EUR"}]}]}}                                    |
+      | BOX1    | [{"type": "add_data", "field": "price", "data": [{"data": 5.3, "currency": "EUR"}], "locale": "fr_FR", "scope": null}]    | {"values": {"price": [{"locale": "fr_FR", "scope": null, "value": [{"data": 5.3, "currency": "EUR"}]}]}}                                  |
+      | BOX1    | [{"type": "add_data", "field": "price", "data": [{"data": "5", "currency": "USD"}], "locale": "fr_FR", "scope": null}]    | {"values": {"price": [{"locale": "fr_FR", "scope": null, "value": [{"data": 5, "currency": "USD"}, {"data": 5.3, "currency": "EUR"}]}]}}  |
```
Comment (sumbobyboys) [inline features/update/add_price.feature]: I'm not a big fan of tests depending on previous lines

File: features/update/add_price.feature [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    And the following attributes:
+      | code   | type   | localizable | scopable |
+      | price  | prices | yes         | no       |
+    And the following products:
+      | sku  |
+      | BOX1 |
+    Then I should get the following products after apply the following updater to it:
+      | product | actions                                                                                                                   | result                                                                                                                                    |
+      | BOX1    | [{"type": "add_data", "field": "price", "data": [{"data": "12.4", "currency": "EUR"}], "locale": "fr_FR", "scope": null}] | {"values": {"price": [{"locale": "fr_FR", "scope": null, "value": [{"data": 12.4, "currency": "EUR"}]}]}}                                 |
+      | BOX1    | [{"type": "add_data", "field": "price", "data": [{"data": 5, "currency": "EUR"}], "locale": "fr_FR", "scope": null}]      | {"values": {"price": [{"locale": "fr_FR", "scope": null, "value": [{"data": 5, "currency": "EUR"}]}]}}                                    |
```
Comment (sumbobyboys) [inline features/update/add_price.feature]: You should also try two adds on the same line to see what happen if there is no flush between two adds

File: spec/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdderSpec.php [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        );
+    }
+
+    function it_is_an_adder()
+    {
+        $this->shouldImplement('Pim\Bundle\CatalogBundle\Updater\Adder\AdderInterface');
+    }
+
+    function it_supports_price_collection_attributes(
+        AttributeInterface $price_collectionAttribute,
```
Comment (sumbobyboys) [inline spec/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdderSpec.php]: camelCase ?

File: features/update/add_price.feature (hunk 2 above)
Comment (willy-ahva) [inline features/update/add_price.feature]: :+1:

File: spec/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdderSpec.php [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        AttributeInterface $textareaAttribute
+    ) {
+        $price_collectionAttribute->getAttributeType()->willReturn('pim_catalog_price_collection');
+        $this->supportsAttribute($price_collectionAttribute)->shouldReturn(true);
+
+        $textareaAttribute->getAttributeType()->willReturn('pim_catalog_textarea');
+        $this->supportsAttribute($textareaAttribute)->shouldReturn(false);
+    }
+
+    function it_checks_locale_and_scope_when_setting_an_attribute_data(
```
Comment (sumbobyboys) [inline spec/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdderSpec.php]: adding, no ? Please avoid to copy and paste :)

File: src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+    /**
+     * Check if data are valid
+     *
+     * @param AttributeInterface $attribute
+     * @param mixed              $data
+     *
+     * @return mixed
+     */
+    protected function checkData(AttributeInterface $attribute, $data)
```
Comment (solivier) [inline src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php]: I think you can split this method

File: src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+     * @param CurrencyManager          $currencyManager
+     * @param array                    $supportedTypes
+     */
+    public function __construct(
+        ProductBuilderInterface $productBuilder,
+        AttributeValidatorHelper $attrValidatorHelper,
+        CurrencyManager $currencyManager,
+        array $supportedTypes
+    ) {
+        parent::__construct($productBuilder, $attrValidatorHelper);
```
Comment (sumbobyboys) [inline src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php]: please add a blank line after

Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php]: no point of splittting it

File: src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php (hunk 5 above)
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php]: no point of splitting it

File: src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php (hunk 6 above)
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Updater/Adder/PriceCollectionAttributeAdder.php]: done

**Pull request merged**